File: Leetcode Problems/Array & String/Merge_Sorted_Array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int i = m;
        for (int j=0; j<n; j++){
            nums1[i] = nums2[j];
            i++;
        }
        Arrays.sort(nums1);
    }
     public static void main(String[] args){
        int []nums1 =  {1, 2, 3, 0, 0, 0}
        int []nums1 =  {2, 4, 5}
        merge(nums1, num2)
        System.out.println(nums1)
        System.out.println(nums2)
    }
}

File: LeetCode Solution Scala/Array and String/Merge_Sorted_Array.scala
object Solution {
    def merge(nums1: Array[Int], m: Int, nums2: Array[Int], n: Int): Unit = {
        var p1 = m-1
        var p2 = n-1
        var p = m + n -1
        while (p2 >= 0){
            if (p1 >= 0 && nums1(p1) > nums2(p2)){
                nums1(p) = nums1(p1)
                p-=1
                p1-=1
            }
            else{
                nums1(p) = nums2(p2)
                p-=1
                p2-=1
            }
        }   
    }
}